dags/taar_amodump.py

```diff
@@ -1,83 +1,127 @@
-    email=["mlopatka@mozilla.com", "vng@mozilla.com"],
-    env=mozetl_envvar("taar_amowhitelist",
-                      {},
-                      {'MOZETL_SUBMISSION_METHOD': 'spark'}),
-    uri="https://raw.githubusercontent.com/mozilla/python_mozetl/master/bin/mozetl-submit.sh",
-    output_visibility="private",
-    dag=dag
+    email=["mlopatka@mozilla.com", "vng@mozilla.com", "hwoo@mozilla.com"],
+    # We are extracting addons from the AMO server's APIs which don't
+    # support date based queries, so no date parameter is required
+    # here.
+    arguments=["-m", "taar_etl.taar_amowhitelist"],
+    env_vars={
+        "AWS_ACCESS_KEY_ID": aws_access_key,
+        "AWS_SECRET_ACCESS_KEY": aws_secret_key,
+    },
+    dag=dag,
 )
 
-editorial_whitelist = EMRSparkOperator(
+editorial_whitelist = GKEPodOperator(
     task_id="taar_update_whitelist",
-    job_name="Generate a JSON blob from editorial reviewed addons for TAAR",
-    execution_timeout=timedelta(hours=1),
-    instance_count=1,
+    project_id=connection.project_id,
+    location="us-central1-a",
+    cluster_name="bq-load-gke-1",
+    name="taar-update-whitelist",
+    namespace="default",
+    # This uses a circleci built docker image from github.com/mozilla/taar_gcp_etl
+    image="gcr.io/moz-fx-data-airflow-prod-88e0/taar_gcp_etl:0.1",
     owner="vng@mozilla.com",
-    email=["mlopatka@mozilla.com", "vng@mozilla.com"],
-    env=mozetl_envvar("taar_update_whitelist",
-                      {"date": "{{ ds_nodash }}"},
-                      {'MOZETL_SUBMISSION_METHOD': 'spark'}),
-    uri="https://raw.githubusercontent.com/mozilla/python_mozetl/master/bin/mozetl-submit.sh",
-    output_visibility="private",
-    dag=dag
+    email=["mlopatka@mozilla.com", "vng@mozilla.com", "hwoo@mozilla.com"],
+    arguments=["-m", "taar_etl.taar_update_whitelist", "--date", "{{ ds_nodash }}"],
+    env_vars={
+        "AWS_ACCESS_KEY_ID": aws_access_key,
+        "AWS_SECRET_ACCESS_KEY": aws_secret_key,
+    },
+    dag=dag,
 )
 
-taar_lite = EMRSparkOperator(
+
+taar_lite = SubDagOperator(
     task_id="taar_lite",
-    job_name="Generate GUID coinstallation JSON for TAAR",
-    instance_count=5,
-    execution_timeout=timedelta(hours=4),
-    owner="mlopatka@mozilla.com",
-    email=["mlopatka@mozilla.com", "vng@mozilla.com"],
-    env=mozetl_envvar("taar_lite",
-                      {"date": "{{ ds_nodash }}"},
-                      {'MOZETL_SUBMISSION_METHOD': 'spark'}),
-    uri="https://raw.githubusercontent.com/mozilla/python_mozetl/master/bin/mozetl-submit.sh",
-    output_visibility="private",
-    dag=dag
+    subdag=moz_dataproc_pyspark_runner(
+        parent_dag_name="taar_amodump",
+        dag_name="taar_lite",
+        default_args=default_args,
+        cluster_name=taarlite_cluster_name,
+        job_name="TAAR_Lite_GUID_GUID",
+        python_driver_code="gs://moz-fx-data-prod-airflow-dataproc-artifacts/jobs/taar_lite_guidguid.py",
+        num_workers=15,
```

Comment:
@haroldwoo @mreid-moz is this ok?  I bumped the number of workers up to 15 so that the job completes in ~30 minutes.  Is there a cost optimization that I should think about?  This job needs to run daily.

Reply:
Should be fine. Does this job take roughly twice the time with twice as many workers?